api/envoy/data/cluster/v3/outlier_detection_event.proto

```diff
@@ -144,3 +150,11 @@ message OutlierEjectFailurePercentage {
   // Host's success rate at the time of the ejection event on a 0-100 range.
   uint32 host_success_rate = 1 [(validate.rules).uint32 = {lte: 100}];
 }
+
+message OutlierEjectExtension {
+  // Name of extension monitor which caused a host ejection.
+  string monitor_name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Any extra info an extension monitor wants to add when a host is ejected.
+  string extra_info = 2;
```

Comment:
Thanks for you great works. :) One comment to the API change.

`extra_info` is a convenient way. but it is like a form of lazy behavior. Seems we don't know what should be record here, so, an omnipotent field is kept here. But it's actually has no any specific meaning.

I am fine to add something like that. But I think at least we should has another field to record the actually event that provided by the monitor, clearly and specifically?

Reply:
Thanks for checking the logic. `extra_info` will be used to log that "extra" information. As per today, I only implemented consecutive errors monitor, so everything is rather clear and no extra info is required. If a host was marked unhealthy, it means that 3 consecutive requests to upstream returned errors. However, for other monitors a bit more information may be beneficial. For example, if `failure percentage` monitor is configured for say 50% and the node is marked unhealthy, the only fact which is known is that percentage of failures was above 50%. The percentage however could be 70%, 80%, 90%, etc. So that `extra_info` field will be used to log actual percentage failure. The same logic will aply to `success_rate` monitor.